pkg/model/serializer_test.go

```diff
@@ -424,26 +421,26 @@ func TestSecretMarshalling(t *testing.T) {
 		{
 			name:          "too-short",
 			data:          "local",
-			expected:      nil,
-			expectedError: true,
+			expected:      &Secret{LocalPath: "", RemotePath: "", Mode: 0},
+			expectedError: false,
 		},
 		{
 			name:          "too-long",
 			data:          "local:remote:mode:other",
-			expected:      nil,
-			expectedError: true,
+			expected:      &Secret{LocalPath: "", RemotePath: "", Mode: 0},
+			expectedError: false,
 		},
 		{
 			name:          "wrong-local",
 			data:          "/local:/remote:400",
-			expected:      nil,
-			expectedError: true,
+			expected:      &Secret{LocalPath: "/local", RemotePath: "/remote", Mode: 256},
+			expectedError: false,
 		},
 		{
 			name:          "wrong-remote",
 			data:          fmt.Sprintf("%s:remote", file.Name()),
-			expected:      nil,
-			expectedError: true,
+			expected:      &Secret{LocalPath: file.Name(), RemotePath: "remote", Mode: 420},
+			expectedError: false,
 		},
 		{
 			name:          "wrong-mode",
```

Comment:
should we fail here and not it validation for a wrong mode?

Reply:
because validation is done outside, a wrong mode will cause the local and remote port to be `nil` only, so it does not fail. We fail when we call `secret.Validate()`